internal/pkg/source/source.go

```diff
@@ -100,47 +98,68 @@ func (a *Source) record(target ssa.Instruction) {
 	}
 }
 
-func (a *Source) visitReferrers(referrers *[]ssa.Instruction) {
-	for _, r := range *referrers {
+func (a *Source) visitReferrers(n ssa.Node) {
+	referrers := a.referrersToVisit(n)
+
+	for _, r := range referrers {
 		// If the referrer is in a different block from the one we last visited,
 		// and it can't be reached from the block we are visiting, then stop visiting.
 		if rb := r.Block(); a.lastBlockVisited != nil && !a.canReach(a.lastBlockVisited, rb) {
 			continue
 		}
-
 		if a.marked[r.(ssa.Node)] {
 			continue
 		}
+
 		switch v := r.(type) {
 		case *ssa.Call:
-			// This is to avoid attaching calls where the source is the receiver, ex:
-			// core.Sinkf("Source id: %v", wrapper.Source.GetID())
-			if recv := v.Call.Signature().Recv(); recv != nil && a.config.IsSource(utils.Dereference(recv.Type())) {
+			if a.config.IsSanitizer(v) {
+				a.sanitizers = append(a.sanitizers, &sanitizer.Sanitizer{Call: v})
+			}
+
+		case *ssa.FieldAddr:
+			if !a.config.IsSourceFieldAddr(v) {
 				continue
 			}
+		}
 
-			if a.config.IsSanitizer(v) {
-				a.sanitizers = append(a.sanitizers, &sanitizer.Sanitizer{Call: v})
+		a.dfs(r.(ssa.Node))
+	}
+}
+
+func (a *Source) referrersToVisit(n ssa.Node) (referrers []ssa.Instruction) {
+	if n.Referrers() == nil {
+		return
+	}
+	for _, r := range *n.Referrers() {
+		// If the referrer is in a different block from the one we last visited,
+		// and it can't be reached from the block we are visiting, then stop visiting.
+		if rb := r.Block(); a.lastBlockVisited != nil &&
+			rb != a.lastBlockVisited &&
+			!a.canReach(a.lastBlockVisited, rb) {
+			continue
```

Comment:
This subsumes the check in `visitReferrers`, which is not unreached by test code.  (I can't comment there, though, since the line is unchanged.  Thanks GitHub.)

Reply:
Good catch! I've removed the check from `visitReferrers`.